docs/marketplace/dev/front-end/ui-components-library/actions/index.md

```diff
@@ -0,0 +1,109 @@
+---
+title: Actions
+description: This article provides details about the Actions service in the Components Library.
+template: concept-topic-template
+---
+
+This article provides details about the Actions service in the Components Library.
+
+## Overview
+
+Actions responsible to handle specific actions that conform to special format within specific context (from Table, Overlay, Http Response, etc.) via Action Handlers.
+This allows backend systems to control some aspects of UI without the need to change frontend at all (ex. update table, close drawer, etc.).
+
+The context within which Actions are handled is defined by the invoker of the Action (Table, Button, Http, etc).
+
+```html
+<spy-button-action
+  [action]="{ 
+    type: 'close-drawer' 
+  }"
+>
+  ...
+</spy-button-action>
+```
+
+## Main Service
+
+Actions is an Angular Service that implements a specific interface (`ActionHandler`) and is registered to the Action Module via `ActionModule.withActions()`. 
+The main service injects all registered types from the `ActionTypesToken`.
+Trigger method finds specific service from the `ActionTypesToken` by `config.type` (from the argument) and returns observable with data by `ActionHandler.handleAction`.
+
+## Action Handler
+
+Actions must implement a specific interface (`ActionHandler`) and then be registered to the Root Module via `ActionModule.withActions()`.
+Action Handler encapsulates the algorithm of how the data is loaded into the Component.
+
+```ts
+// Module augmentation
+import { ActionConfig } from '@spryker/actions';
+
+declare module '@spryker/actions' {
+  interface ActionsRegistry {
+    custom: CustomActionHandlerService;
+  }
+}
+
+export interface CustomActionConfig extends ActionConfig {
+  data: unknown;
+  ...;
+}
+
+// Service implementation
+@Injectable({
+  providedIn: 'root',
+})
+export class CustomActionHandlerService implements ActionHandler {
+  trigger<C extends ActionConfig>(
```

Comment:
Please use the correct `handleAction` method.

Reply:
This is an example of how to create a custom method, you can check already posted articles (for ex. `datasources`)